active/0028-durable-shared-subscriptions.md

```diff
@@ -18,119 +27,209 @@ Since we have durable storage-based implementation for the regular subscriptions
 
 ### General Idea
 
-Each group is managed by a unique shared group leader process (SGL) that is cluster-wide. It has a persistent state but does not always run. It spawns when any group consumer connects and exits for some time after the last consumer disappears.
+Shared subscriptions (or queues) are a feature that allows multiple consumers to consume messages from some topic filter cooperatively.
 
-An SGL is spawned by a shared group leader manager (SGLM) process. SGLM runs per node and is responsible for spawning SGLs, locating them in the cluster, and guaranteeing that there is only one SGL per group.
+Several consumers subscribe to a special topic `$shared/GROUP_ID/TOPIC_FILTER` (or `$queue/GROUP_ID/TOPIC_FILTER` in the case of queues) and consume messages from `TOPIC_FILTER`. Each message goes to a single consumer.
 
-The global registration mechanism has not yet been chosen. It may be `ra`-based, `global`-based, or even DS-specific (like registering in FoundationDB).
+With the durable backend (Durable Storage backend, DS), all messages pertain to the ordered _streams_ of messages. Streams may be read sequentially (possibly skipping some messages). Streams may be finalized (i.e., closed) and so fully consumed.
 
-SGL keeps track of topics belonging to the group, their streams, and stream states. It is the only entity that tracks the replay progress of these streams.
+Since the streams are a means of sharding messages, the natural idea is to use the same sharding mechanism for shared subscriptions. That is, assign disjointed subsets of streams to different consumers and let them consume their streams in parallel.
```

Comment:
```suggestion
Since the streams are means of sharding messages, the natural idea is to use the same sharding mechanism for shared subscriptions. That is, assign disjointed subsets of streams to different consumers and let them consume their streams in parallel.
```